### ISSUE TITLE
Slash in path breaks other paths in brace syntax

### ISSUE THREAD

**Issue by poconn** [editorconfig/editorconfig-core-js#118]
In a `[{pattern1,pattern2,...}]` section, a slash in one pattern seems to cause the others to only match files in the project root directory instead of including subdirectories:

Matches all `**/*.ext`:
```
[{*.ext,foo}]
```
Only matches `/*.ext`:
```
[{*.ext,foo/bar}]
```

Version:
```bash
$ editorconfig --version
EditorConfig Node.js Core Version 1.0.2
```

Thanks!

**Comment by hildjj** [comment]
Confirmed.  I think this may be an issue in [minimatch](https://github.com/isaacs/minimatch).  If this is still important to you, would you mind filing an issue over there?  Some specifics of the test I just ran:

- minimatch v9.0.1
- `minimatch('/tmp/boo/bar.ext', '/tmp/{*.ext,foo/bar}', { matchBase: true, dot: true, noext: true })` => false

`matchBase: false` doesn't change anything here.

**Comment by hildjj** [comment]
`noext: false` also doesn't change this.

**Comment by hildjj** [comment]
I don't think this is a minimatch issue.  I think it may be an issue with the `buildFullGlob` routine.

**Comment by hildjj** [comment]
On further reading of the spec, I think this may work as intended.  From [version 0.17.2](https://spec.editorconfig.org/#glob-expressions):

```
If the glob contains a path separator (a / not inside square brackets), then the glob is relative to the directory level of the particular .editorconfig file itself. Otherwise the pattern may also match at any level below the .editorconfig level. For example, *.c matches any file that ends with .c in the directory of .editorconfig or any other directory below one that stores this .editorconfig. However, the glob subdir/*.c only matches files that end with .c in the subdir directory in the directory of .editorconfig.
```

Therefore, the workaround of using `[{**/*.ext,foo/bar}]` is probably the right way to go.
